### ISSUE TITLE
crash when using multiple processors and missing variables

### ISSUE THREAD

**Issue by oloapinivad** [oloapinivad/ECmean4#54]
We found out that if there are more available processors than variable to be processed the code might crash. 
This is currently investigated.

**Comment by oloapinivad** [comment]
Here the traceback of the case when the problem show up.
```

Error 11: Resource temporarily unavailable
Process Process-12:
Traceback (most recent call last):
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/paolo/ECmean4/./performance_indices.py", line 224, in pi_worker
    out = slicearray.weighted(weights).mean().values
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/dataarray.py", line 729, in values
    return self.variable.values
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/variable.py", line 608, in values
    return _as_array_or_item(self._data)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/variable.py", line 314, in _as_array_or_item
    data = np.asarray(data)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/array/core.py", line 1700, in __array__
    x = self.compute()
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/base.py", line 315, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/base.py", line 600, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 557, in get_sync
    return get_async(
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 500, in get_async
    for key, res_info, failed in queue_get(queue).result():
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 542, in submit
    fut.set_result(fn(*args, **kwargs))
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 238, in batch_execute_tasks
    return [execute_task(*a) for a in it]
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 238, in <listcomp>
    return [execute_task(*a) for a in it]
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 229, in execute_task
    result = pack_exception(e, dumps)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/local.py", line 224, in execute_task
    result = _execute_task(task, data)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/optimization.py", line 990, in __call__
    return core.get(self.dsk, self.outkey, dict(zip(self.inkeys, args)))
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/core.py", line 149, in get
    result = _execute_task(task, cache)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/dask/array/core.py", line 126, in getter
    c = np.asarray(c)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/indexing.py", line 459, in __array__
    return np.asarray(self.array, dtype=dtype)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/indexing.py", line 623, in __array__
    return np.asarray(self.array, dtype=dtype)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/indexing.py", line 524, in __array__
    return np.asarray(array[self.key], dtype=None)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/coding/variables.py", line 72, in __array__
    return self.func(self.array)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/coding/variables.py", line 139, in _apply_mask
    data = np.asarray(data, dtype=dtype)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/indexing.py", line 524, in __array__
    return np.asarray(array[self.key], dtype=None)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/backends/netCDF4_.py", line 85, in __getitem__
    return indexing.explicit_indexing_adapter(
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/core/indexing.py", line 815, in explicit_indexing_adapter
    result = raw_indexing_method(raw_key.tuple)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/site-packages/xarray/backends/netCDF4_.py", line 98, in _getitem
    array = getitem(original_array, key)
  File "src/netCDF4/_netCDF4.pyx", line 4973, in netCDF4._netCDF4.Variable.__getitem__
  File "src/netCDF4/_netCDF4.pyx", line 5930, in netCDF4._netCDF4.Variable._get
  File "src/netCDF4/_netCDF4.pyx", line 2028, in netCDF4._netCDF4._ensure_nc_success
RuntimeError: Resource temporarily unavailable
Traceback (most recent call last):
  File "/home/paolo/ECmean4/./performance_indices.py", line 437, in <module>
    sys.exit(pi_main(sys.argv[1:]))
  File "/home/paolo/ECmean4/./performance_indices.py", line 378, in pi_main
    ordered[item] = varstat[item]
  File "<string>", line 2, in __getitem__
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/multiprocessing/managers.py", line 825, in _callmethod
    raise convert_to_error(kind, result)
  File "/work/users/paolo/miniconda3/envs/ECmean4/lib/python3.9/multiprocessing/managers.py", line 265, in serve_client
    res = function(*args, **kwds)
KeyError: 'tos'
```

**Comment by oloapinivad** [comment]
Apparently this is solved by a small change in #68 which load the climatology before looping
